🌳 Feature - Enhance plugin to include "deletedId" on deleted docs (#2)
Lighthouse-io/mongoose-delete: README.md

  - [Add __deleted__ (true-false) key on document](#simple-usage)
  - [Add __deletedAt__ key to store time of deletion](#save-time-of-deletion)
  - [Add __deletedBy__ key to record who deleted document](#who-has-deleted-the-data)

>   - [Add __deletedId__ key to store reference to action/event that caused deletion](#track-deletion-action-id)

Reviewer: 🤏 I'd mention that the main use of this is to track when you're deleting multiple documents as part of a single action
```suggestion
  - [Add __deletedId__ key to record operation that included this deletion](#track-deletion-action-id)
```

});
```

### Track deletion action ID

The `deletedId` can be used to track items that were deleted as part of a single operation.

```javascript
var mongoose_delete = require('mongoose-delete');

var PetSchema = new Schema({
    name: String
});

PetSchema.plugin(mongoose_delete, { deletedId: true });

var Pet = mongoose.model('Pet', PetSchema);

async function example() {
  try {
    // Create two pets
    const fluffy = new Pet({ name: 'Fluffy' });
    const bitey = new Pet({ name: 'Bitey' });

    // Create a shared action ID for grouping the deletions
    const actionId = new mongoose.Types.ObjectId("61fc42a56b4a6670076b16bf");

    // Delete both pets with the same actionId
    // This allows tracking which documents were deleted in the same operation
    await Promise.all([
      fluffy.delete({ deletedId: actionId }),
      bitey.delete({ deletedId: actionId })
    ]);
    // mongodb: { deleted: true, name: 'Fluffy', deletedId: ObjectId("61fc42a56b4a6670076b16bf")}
    // mongodb: { deleted: true, name: 'Bitey', deletedId: ObjectId("61fc42a56b4a6670076b16bf")}
    
    // Find all pets deleted as part of this action

>     const deletedPets = await Pet.findWithDeleted({ deletedId: actionId });

Reviewer: 🤏 I think `findWithDeleted` might be a LH function yeah? That probably means you'll need to have your own equivalent implementation in your docs.

Author: `findWithDeleted` is a mongoose-delete function that can be used instead of `find`, so I think this is fine here?

});
```

### Track deletion action ID

> ```javascript

Reviewer: ```suggestion
### Track deletion action ID

The `deletedId` can be used to track items that were deleted as part of a single operation.

```javascript
```

var mongoose_delete = require('mongoose-delete');

var PetSchema = new Schema({
    name: String
});

PetSchema.plugin(mongoose_delete, { deletedId: true });

var Pet = mongoose.model('Pet', PetSchema);

> var fluffy = new Pet({ name: 'Fluffy' });

Reviewer: ```suggestion
var fluffy = new Pet({ name: 'Fluffy' });
var bitey = new Pet({name: 'Bitey' });
```

> fluffy.save(function () {

Reviewer: 🤌 The example doesn't really illustrate what this is doing and is intended for. I'm wondering if you should go with a second collection here, so something like `FoodSchema`, then when you delete the pet you can delete the food at the same time with the id and use both to restore

Author: I've reused the PetSchema twice to demonstrate this, hope this is clear enough but I can add FoodSchema if you think that'd be better

    // mongodb: { deleted: false, name: 'Fluffy' }

    var actionId = mongoose.Types.ObjectId("61fc42a56b4a6670076b16bf");

    // Pass deletedId as an option
    fluffy.delete({ deletedId: actionId }, function () {
        // mongodb: { deleted: true, name: 'Fluffy', deletedId: ObjectId("61fc42a56b4a6670076b16bf")}

        fluffy.restore(function () {
            // mongodb: { deleted: false, name: 'Fluffy' }
        });
    });

> });

Reviewer: 🤔 You'll need to turn this into async/await code for it to make any sense. It'll be a horrible mess of callback code otherwise.
